pull/pull_requests.go

```diff
@@ -26,49 +26,36 @@ import (
 // ListOpenPullRequestsForSHA returns all pull requests where the HEAD of the source branch
 // in the pull request matches the given SHA.
 func ListOpenPullRequestsForSHA(ctx context.Context, client *github.Client, owner, repoName, SHA string) ([]*github.PullRequest, error) {
-	var results []*github.PullRequest
-
-	openPRs, err := ListOpenPullRequests(ctx, client, owner, repoName)
-
+	prs, _, err := client.PullRequests.ListPullRequestsWithCommit(ctx, owner, repoName, SHA, &github.ListOptions{
+		// In practice, there should be at most 1-3 PRs for a given commit. In
+		// exceptional cases, if there are more than 100 PRs, we'll only
+		// consider the first 100 to avoid paging.
+		PerPage: 100,
+	})
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to list pull requests for repository %s/%s", owner, repoName)
 	}
 
-	for _, openPR := range openPRs {
-		if openPR.Head.GetSHA() == SHA {
-			results = append(results, openPR)
+	var results []*github.PullRequest
+	for _, pr := range prs {
+		if pr.GetState() == "open" && pr.GetHead().GetSHA() == SHA {
+			results = append(results, pr)
 		}
 	}
-
 	return results, nil
 }
 
 func ListOpenPullRequestsForRef(ctx context.Context, client *github.Client, owner, repoName, ref string) ([]*github.PullRequest, error) {
 	var results []*github.PullRequest
 	logger := zerolog.Ctx(ctx)
 
-	openPRs, err := ListOpenPullRequests(ctx, client, owner, repoName)
-
-	if err != nil {
-		return nil, err
+	if strings.HasPrefix(ref, "refs/heads/") {
+		ref = strings.TrimPrefix(ref, "refs/heads/")
```

Comment:
nit: `TrimPrefix` returns the unmodified input if it doesn't contain the prefix, so there's usually no need to test with `HasPrefix` before calling `TrimPrefix. If you do care about knowing if the prefix existed or not, you can also use `CutPrefix`.